pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiWorkflowState.java

```diff
@@ -36,9 +36,9 @@ int insert(@Bind("stateId") long stateId, @Bind("studyGuid") String studyGuid,
             + " where study_activity_id = :activityId and check_each_instance = :check")
     Optional<Long> findActivityStateId(@Bind("activityId") long activityId, @Bind("check") boolean checkEachInstance);
 
-    @SqlQuery("select workflow_state_id, study_guid, study_name, redirect_url from workflow_study_redirect_state "
-            + " where (study_guid is null or study_guid = :studyGuid) and (study_name is null or study_name = :studyName) "
-            + " and redirect_url = :redirectUrl")
+    @SqlQuery("select * from workflow_study_redirect_state"
+            + " where (study_guid = :studyGuid and redirect_url = :redirectUrl)"
+            + "    or (study_name = :studyName and redirect_url = :redirectUrl)")
```

Comment:
The old query was not working for me (the add-child workflow adds same study redirect rules and it's creating duplicate rows). No idea why but this new query seems to work.

Reply:
@yufengwng was it create duplicates in unit tests or in dev env? May be somehow affects DB data that duplicated rows were created?

Reply:
I think it somehow couldn't find the existing row in DB and tried to create another one and causing duplicate error, not sure why it couldn't find it though.